taipy/gui/builder/_api_generator.py

```diff
@@ -139,5 +139,9 @@ def create_element_api(
         return type(
             classname,
             (ElementBaseClass,),
-            {"_ELEMENT_NAME": element_name, "_DEFAULT_PROPERTY": default_property, "_TYPES": properties},
+            {
+                "_ELEMENT_NAME": element_name,
+                "_DEFAULT_PROPERTY": default_property,
+                "_TYPES": {f"{parts[0]}__" if len(parts := k.split("[")) > 1 else k: v for k, v in properties.items()},
```

Comment:
I dont understand why do we split by `[` here.

Reply:
So that we can access with a get the indexed property name and check if it's callable